src/de.hpi.swa.trufflesqueak/src/de/hpi/swa/trufflesqueak/nodes/ExecuteBytecodeNode.java

```diff
@@ -54,6 +55,10 @@ public Object execute(final VirtualFrame frame, final int startPC) {
             return interpretBytecode(frame, startPC);
         } catch (final NonLocalReturn nlr) {
             /* {@link getHandleNonLocalReturnNode()} acts as {@link BranchProfile} */
+            if (FrameAccess.isDead(frame)) {
+                LogUtils.SCHEDULING.fine("ExecuteBytecodeNode: encountered dead frame during NLR");
+                throw nlr;
+            }
             return getHandleNonLocalReturnNode().executeHandle(frame, nlr);
```

Comment:
The comment means that the lazy-initialized node is used as a `BranchProfile` to hide the code path if it has never been taken for a specific method. The new code here, however, is before the lazy-init getter. To hide it also, you have to do something like this instead:

```suggestion
            /* {@link getHandleNonLocalReturnNode()} acts as {@link BranchProfile} */
            final HandleNonLocalReturnNode handleNode = getHandleNonLocalReturnNode();
            if (FrameAccess.isDead(frame)) {
                LogUtils.SCHEDULING.fine("ExecuteBytecodeNode: encountered dead frame during NLR");
                throw nlr;
            }
            return handleNode.executeHandle(frame, nlr);
```

Hope this makes sense :)